pydicom/filereader.py

```diff
@@ -1008,22 +1021,41 @@ def read_deferred_data_element(fileobj_type, filename_or_obj, timestamp,
           if is_filename else filename_or_obj)
     is_implicit_VR = raw_data_elem.is_implicit_VR
     is_little_endian = raw_data_elem.is_little_endian
+
+    if config._memmap_size and raw_data_elem.length > config._memmap_size:
```

Comment:
Should also check for binary VR here as above (or move the check into a variable).